packages/zoe/src/contracts/coveredCall.js

```diff
@@ -1,84 +1,94 @@
+ *    time, timer: myTimer }
+ *    },
  * }
- * The result of the initial offer is a seat where the payout will
- * eventually resolve either to the strikePrice or to a refund of the
- * underlying asset. The offerResult is the call option, which is an
- * assayable invitation to buy the underlying asset. Since the
- * contract provides assurance that the underlying asset is available
- * on the specified terms, the invitation itself can be traded as a
- * valuable good.
+ *
+ * This deadline serves as the expiration date for the covered call
+ * option. After this deadline, if the option has not been exercised,
+ * the underlying assets are automatically paid out to the creator of
+ * the contract as a refund.
+ *
+ * After the owner of the digital assets escrows the assets in the
+ * initial offer, they receive a seat. The payout for this seat will
+ * either be a refund of the underlying assets (as mentioned above) or
+ * the strike price. Zoe's enforcement of offer safety guarantees that
+ * the payout is either a refund or the strike price, regardless of
+ * whether this contract is buggy.
+ *
+ * The offerResult of this initial seat resolves to the call option
+ * itself: an assayable invitation to buy the underlying assets. The
+ * call option invitation has this additional information in the
+ * value: { expirationDate, timerAuthority, underlyingAssets,
+ * strikePrice }
+ *
+ * The invitation itself can be traded as a valuable digital asset: a
+ * covered call option.
+ *
+ * The recipient of a covered call option (buying it on some other
+ * exchange or through some other trading contract) can exercise the
+ * option before the deadline by using the option as an invitation to
+ * this contract, paying the strike price and receiving the underlying
+ * assets. The recipient of a covered call option can use whatever
+ * keywords they wish, as long as they give the strike price as
+ * specified in the invitation value, and want the underlying assets
+ * exactly.
  *
  * @type {ContractStartFn}
  */
 const start = zcf => {
-  assertIssuerKeywords(zcf, harden(['UnderlyingAsset', 'StrikePrice']));
+  const sellSeatExpiredMsg = `The covered call option is expired.`;
 
-  const makeCallOption = sellerSeat => {
-    assertProposalShape(sellerSeat, {
-      give: { UnderlyingAsset: null },
-      want: { StrikePrice: null },
-      exit: { afterDeadline: null },
-    });
-    const { want, give, exit } = sellerSeat.getProposal();
-    const rejectMsg = `The covered call option is expired.`;
+  const makeOption = sellSeat => {
+    assertProposalShape(sellSeat, { exit: { afterDeadline: null } });
 
-    const exerciseOption = exerciserSeat => {
-      assertProposalShape(exerciserSeat, {
-        give: { StrikePrice: null },
-        want: { UnderlyingAsset: null },
-      });
-      const swapResult = swap(zcf, sellerSeat, exerciserSeat, rejectMsg);
+    const exerciseOption = buySeat => {
+      try {
```

Comment:
This is an uncomfortable pattern for debug info. Partly, try/catch that just adds info clutters the code. We pass one error message in but add this on the outside. should there be another error message argument?  an error object with properties for the various error modes?  This seems like we should discuss also with @erights

Reply:
Expanding on this, these are tutorial messages. If `throw`n errors included codes, then it would be easier to add detailed documentation externally for what/why.

Reply:
Yup, I don't like the awkwardness of this either. But it seems like a long term thing that we should assess.